omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java

```
@@ -0,0 +1,79 @@
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/passwordreset")
+public class PasswordResetController2_2 extends BaseRestController {
+	
+	@Qualifier("userService")
+	@Autowired
+	private UserService userService;
+	
+	@RequestMapping(method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.OK)
+	public void requestPasswordReset(@RequestBody Map<String, String> body) {
+		User user = null;
+		String usernameOrEmail = body.get("usernameOrEmail");
+		if ((user = userService.getUserByEmail(usernameOrEmail)) == null) {
+			user = userService.getUserByUsername(usernameOrEmail);
+		}
+		//user = userService.getUserByEmail(usernameOrEmail);
+		if (user != null) {
+			userService.setUserActivationKey(user);
+		}
+	}
+	
+	@RequestMapping(value = "/{activationkey}", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<String> resetPassword(@PathVariable("activationkey") String activationkey,
+	        @RequestBody Map<String, String> body) {
+		String newPassword = body.get("newPassword");
+		if (userService.getUserByActivationKey(activationkey) == null) {
+			return new ResponseEntity<String>("Invalid Activation Key", HttpStatus.BAD_REQUEST);
```

You should instead throw an exception

---

Possibly a NPE